src/vs/workbench/contrib/tasks/common/taskConfiguration.ts

```diff
@@ -1339,6 +1340,9 @@ namespace ConfiguringTask {
 			));
 			return undefined;
 		}
+		if (identifier.type === 'typescript' && identifier.tsconfig) {
```

Comment:
Having a check for an extension contributed task type in VS Code core breaks our layering philosophy: VS Code core shouldn't know about anything extension specific. Instead, the typescript extension should do this. It should be somewhere around here: https://github.com/Microsoft/vscode/blob/master/extensions/typescript-language-features/src/features/task.ts#L174

Reply:
Yes I was very hesitant of this if condition as well. But in this case, ts extension code is correct and VS Code core has the bug! tsconfig file path normalization is a very special case, non of the other supported task types have a path string in them.
The if condition a few lines above `if (customize.indexOf(typescript) === 0)` inspired me to do this. 

The only alternative I can think of is that, inside ts extention, when tasks are generated, the task label should be read from tasks.json file instead of the auto generated Task. Which means, ts extension should find and read .vscode/tasks.json. I don't think that is also a suggested change!

Reply:
In the typescript extension, can't you just use the OS specific path for the tsconfig?

Reply:
Thinking about it some more, it seems like the typescript extension should just not use the OS specific path for the tsconfig property.

Reply:
For typescript extension to use what's written in the tasks.json instead of OS specific path, we will have to read .vscode/tasks.json from the extension, if that is advisable, I am willing to do that. What do you say?

Reply:
I don't think that the typescript should use what's written in tasks.json, I just think that it should use the non-OS specific path.

Reply:
Means it should always use '/' irrespective of the OS?

Reply:
That is what I meant, but actually we shouldn't change this. I don't want to break existing users who have it configured using the OS specific path. Because of this, there shouldn't be a code fix for this issue.

Reply:
We do have precendent for breaking things like this though. It would be a better experience to use the non-OS specific path in the typescript extension. That is still the change we should make!

Reply:
Alright @alexr00, will make the change accordingly but I think, its a big one. Unless 'tsconfig' is the only property in typescript task which has path as a value. Will update this PR.